_source/_posts/2022-04-06-react-azure-functions.md

```diff
@@ -0,0 +1,416 @@
+---
+layout: blog_post
+title: "How to Build and Deploy a Serverless React App on Azure"
+author: nickolas-fisher
+by: contractor
+communities: [javascript]
+description: "A tutorial that shows you how to build and deploy a React app that securely calls serverless functions on Azure."
+tags: [react, azure, serverless, javascript]
+tweets:
+  - ""
+  - ""
+  - ""
+image:
+type: conversion
+github: https://github.com/oktadev/okta-react-azure-functions-example
+---
+
+Microsoft's Azure platform has as many high-tech products as anyone could ever want. Among these is the [Azure Static Web Apps](https://azure.microsoft.com/en-us/services/app-service/static/). As the name suggests, the platform hosts static web apps that don't require a back end. Azure supports React, Angular, Vue, Gatsby, and many more out of the box.
+
+However, you may run into a situation where you need a little back-end support. You can imagine where you might need the backend to run one or two API calls. For this task, Azure offers the [Functions](https://azure.microsoft.com/en-us/services/functions/) platform as well. Functions is a serverless computing platform that supports .NET, Node.js, Python, etc. It takes setting up a server, building logging and exception handling, and providing a high availability environment at a reasonable price.
+
+This tutorial will show you how to create a React application and deploy it to Azure Static Web Apps. This application will be on the Azure free tier, so you will not be able to rely on the built-in authentication providers that connect Azure and Okta to handle the authentication. Therefore you will use the `okta-react` package from Okta to secure your SPA manually. Once the user authenticates, they may upload an image and receive a badge from a serverless Azure function.
+
+This function will handle the work of accepting the inputted image from the SPA and using a template to create a personalized badge for the user. Since you will be using the free version of Azure Static Web Apps, you will have to deploy the Function as a _Managed Azure Function_.
```

Comment:
```suggestion
This serverless function will handle the work of accepting the input image from the SPA and using a template to create a personalized badge for the user. Since you will be using the free version of Azure Static Web Apps, you will have to deploy the function as a _Managed Azure Function_.
```
inputted is not wrong, but input reads better and works as past tense.